src/main/java/io/kabanero/instance/KabaneroManager.java

```diff
@@ -27,25 +29,21 @@
 
 import io.website.Constants;
 import io.kubernetes.KabaneroClient;
+import io.kubernetes.client.ApiException;
 
 // Singleton class to manage the various kabanero instances associated with Kabanero
 public class KabaneroManager {
     private final static Logger LOGGER = Logger.getLogger(KabaneroManager.class.getName());
 
     private static KabaneroManager SINGLE_KABANERO_MANAGER_INSTANCE;
     private HashMap<String, KabaneroInstance> KABANERO_INSTANCES = new HashMap<String, KabaneroInstance>();
-    private long created = System.currentTimeMillis();
 
     private KabaneroManager() {
     }
 
-    private boolean isOld() {
-        return (System.currentTimeMillis() - SINGLE_KABANERO_MANAGER_INSTANCE.created > 1000 * 60 * 5);
-    }
-    
-    public static synchronized KabaneroManager getKabaneroManagerInstance() {
-        // quick hack: isOld will force refresh every so often - we should be watching for changes instead
-        if(SINGLE_KABANERO_MANAGER_INSTANCE == null || SINGLE_KABANERO_MANAGER_INSTANCE.isOld()) {
+    public static synchronized KabaneroManager getKabaneroManagerInstance() throws IOException, GeneralSecurityException, ApiException {
+
+        if(SINGLE_KABANERO_MANAGER_INSTANCE == null || KabaneroClient.isOld()) {
```

Comment:
this is a good start, you shouldn't need an `isOld()` method anymore. The `SINGLE_KABANERO_MANAGER_INSTANCE ` instance obj should just update itself from the watchers